### PR TITLE
#337 - add dates to medical exams and ohs

### LINKED ISSUE #337: Add dates related to occupational medicine examinations and BHP
The goal of this task is to add dates related to occupational medicine examinations and BHP trainings to each of user. On the users' page, fields such as: date of the last examination, date of the next examination, date of the last BHP training, and date of the next BHP training should be added. Additionally, if a given employee's examination or BHP training date is approaching, an email (or possibly a Slack notification) should be sent to the administrative person in charge two months before and one month before, as a reminder of the deadlines and a list of people.

### PR DESCRIPTION
**Pull request by EwelinaSkrzypacz**
This should close #337.

image

image

### CODE REVIEW

File: app/Eloquent/Models/Profile.php [hunk 1]
```diff
@@ -21,6 +21,10 @@
  * @property EmploymentForm $employment_form
  * @property Carbon $employment_date
  * @property Carbon $birthday
+ * @property Carbon $last_medical_exam_date
```
Comment (krzysztofrewak) [inline app/Eloquent/Models/Profile.php:24]: What about nullables?

File: app/Eloquent/Models/Profile.php [hunk 2]
```diff
@@ -21,6 +21,10 @@
...
+ * @property Carbon $next_medical_exam_date
+ * @property Carbon $last_ohs_training_date
+ * @property Carbon $next_ohs_training_date
```
Comment (krzysztofrewak) [inline app/Eloquent/Models/Profile.php:27]:
```suggestion
 * @property ?Carbon $last_medical_exam_date
 * @property ?Carbon $next_medical_exam_date
 * @property ?Carbon $last_ohs_training_date
 * @property ?Carbon $next_ohs_training_date
```

File: app/Infrastructure/Console/Kernel.php [hunk 3]
```diff
@@ -36,6 +38,10 @@ protected function schedule(Schedule $schedule): void
             ->weekdays()
             ->dailyAt("08:30");
 
+        $schedule->command(SendNotificationAboutUpcomingAndOverdueMedicalExams::class)->monthlyOn(1, "08:00");
```
Comment (krzysztofrewak) [inline app/Infrastructure/Console/Kernel.php:41]:
```suggestion
        $schedule->command(SendNotificationAboutUpcomingAndOverdueMedicalExams::class)->monthlyOn(1, "08:00")->onOneServer();
```

Please change it in other command too: https://laravel.com/docs/10.x/scheduling#running-tasks-on-one-server

File: database/migrations/2023_11_29_142131_add_dates_of_medical_exams_and_ohs_in_profiles_table.php [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+use Illuminate\Support\Facades\Schema;
+
+return new class() extends Migration {
+    public function up(): void
+    {
+        Schema::table("profiles", function (Blueprint $table): void {
+            $table->date("last_medical_exam_date")->nullable();
+            $table->date("next_medical_exam_date")->nullable();
+            $table->date("last_ohs_training_date")->nullable();
+            $table->date("next_ohs_training_date")->nullable();
```
Comment (krzysztofrewak) [inline database/migrations/2023_11_29_142131_add_dates_of_medical_exams_and_ohs_in_profiles_table.php:16]:
```suggestion
            $table->date("last_medical_exam_date")->nullable()->default(null);
            $table->date("next_medical_exam_date")->nullable()->default(null);
            $table->date("last_ohs_training_date")->nullable()->default(null);
            $table->date("next_ohs_training_date")->nullable()->default(null);
```

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

**Review by krzysztofrewak — APPROVED** [review]

**Review by Baakoma — APPROVED** [review]

**Pull request merged**